lib/go/thrift/application_exception.go

```diff
@@ -59,9 +62,35 @@ type TApplicationException interface {
 	Write(ctx context.Context, oprot TProtocol) error
 }
 
+type ValidationError struct {
+	message     string
+	check       string
+	fieldSymbol string
+}
+
+func (e *ValidationError) Check() string {
+	return e.check
+}
+
+func (e *ValidationError) Field() string {
+	if fs := strings.Split(e.fieldSymbol, "."); len(fs) > 1 {
```

Comment:
why is this `Split` needed? in which case would we use `.` to construct field names/symbols?

Reply:
`fieldSymbol` indicates a string like 'BasicTest.Double1', which means that the field is the 'Double1' in type 'BasicTest'. And similarly 'BasicTest.Map0.key' indicates that the field is the key of field 'Map0' in type 'BasicTest'.
I also add a new method 'TypeName' to ValidationError to get the name of the type.

Reply:
why not just return the whole string, `BasicTest.Map0.key`? now we have 2 separated APIs to return `BasicTest` and `Map0`, what about the `key` part? this makes things unnecessarily complicated.

or just export the fields and remove the getters. we don't really need getters here anyways, if someone really want to mutate the error returned, they are on their own risks.